netmiko/cisco/cisco_xr.py

```diff
@@ -105,36 +114,74 @@ def commit(
-            strip_prompt=False,
-            strip_command=False,
-            read_timeout=read_timeout,
-        )
-        if "onfirm" in new_data:
-            output += new_data
+        if replace:
+            new_data = self._send_command_str(
+                command_string,
+                expect_string=r"This commit will replace or remove the entire running configuration",
+                strip_prompt=False,
+                strip_command=False,
+                read_timeout=read_timeout,
+            )
+            if "This commit will replace or remove the entire running configuration" in new_data:
+                output += new_data
+                new_data = self._send_command_str(
+                    "yes",
+                    expect_string=r"#",
+                    strip_prompt=False,
+                    strip_command=False,
+                    read_timeout=read_timeout,
+                )
+        else:
+            # IOS-XR might do this:
+            # This could be a few minutes if your config is large. Confirm? [y/n][confirm]
             new_data = self._send_command_str(
-                "y",
-                expect_string=r"#",
+                command_string,
+                expect_string=r"(#|onfirm)",
                 strip_prompt=False,
                 strip_command=False,
                 read_timeout=read_timeout,
             )
+            if "onfirm" in new_data:
+                output += new_data
+                new_data = self._send_command_str(
+                    "y",
+                    expect_string=r"#",
+                    strip_prompt=False,
+                    strip_command=False,
+                    read_timeout=read_timeout,
+                )
         output += new_data
         if error_marker in output:
             raise ValueError(f"Commit failed with the following errors:\n\n{output}")
         if alt_error_marker in output:
+            if commit_error_dialog_dict is not None and alt_error_marker in commit_error_dialog_dict:
+                marker_value = commit_error_dialog_dict[alt_error_marker]
+                output += new_data
+                new_data += self._send_command_str(
+                    marker_value,
+                    expect_string=r"\)#$",
+                    strip_prompt=False,
+                    strip_command=False,
+                    read_timeout=read_timeout,
+                )
+                output += new_data
+            else:
             # Other commits occurred, don't proceed with commit
-            output += self._send_command_timing_str(
-                "no", strip_prompt=False, strip_command=False
-            )
-            raise ValueError(f"Commit failed with the following errors:\n\n{output}")
+                output += self._send_command_timing_str(
+                    "no", strip_prompt=False, strip_command=False
+                )
+                raise ValueError(f"Commit failed with the following errors:\n\n{output}")
 
```

Comment:
print a log here. log.debug("commit Output: {0}".format(output))
should add in all major methods, send_command, enter_config_mode, check_config_mode, commit, exit_config_mode, etc.